main.tf

```diff
@@ -263,11 +263,10 @@ resource "aws_s3_bucket_replication_configuration" "default" {
       }
 
       dynamic "source_selection_criteria" {
-        for_each = try(rule.value.source_selection_criteria.sse_kms_encrypted_objects.enabled, null) == null ? [] : [rule.value.source_selection_criteria.sse_kms_encrypted_objects.enabled]
-
+        for_each = try(rule.value.source_selection_criteria.sse_kms_encrypted_objects.status, "") == "Enabled" ? [1] : []
```

Comment:
@oscar123mendoza Since the type of `s3_replication_rules ` is `list(any)`, can you not pass in `"Enabled"` as the value at this path (`rule.value.source_selection_criteria.sse_kms_encrypted_objects.enabled`) and it'll accomplish the same thing?

Can you please expand more in the Pull Request description or here via a reply on why this is needed? Is this a nicety to make it so you can pass a boolean to that value or something more?

Thanks!